nixos/modules/services/web-apps/nextcloud.nix

```diff
@@ -1002,6 +1020,19 @@ in {
               exit 1
             fi
 
+            ${optionalString (!cfg.disablePathSafetyCheck) ''
+              # Check if systemd-tmpfiles setup worked correctly
+              for i in config data; do
+                if [[ ! -O "${datadir}/$i" ]]; then
+                  echo "${datadir}/$i is not owned by user 'nextcloud'!"
```

Comment:
As for the data/ directory it seems only PHP's `is_writable` is used to check if the directory is "OK": https://github.com/nextcloud/server/blob/fdc0b1ecf423e400257edf868a51d5873fc66c64/lib/private/Setup.php#L248

So here it could totally be owned by another user, however there is another check that ensures the permissions are 770 at max: https://github.com/nextcloud/server/blob/fdc0b1ecf423e400257edf868a51d5873fc66c64/lib/private/legacy/OC_Util.php#L596

So some more fine-grained checks would be better here.